exp/controllers/rosaroleconfig_controller.go

```diff
@@ -0,0 +1,663 @@
+	}
+
+	createRoles := true
+	accountRoles, err := runtime.AWSClient.ListAccountRoles(config.Version)
+	if err != nil {
+		// Let create account roles continue if no account roles are found
+		if !strings.Contains(err.Error(), "no account roles found") {
+			return err
+		}
+	}
+
+	for _, role := range accountRoles {
+		if role.RoleName == fmt.Sprintf("%s-HCP-ROSA-Installer-Role", config.Prefix) {
+			createRoles = false
+			scope.RosaRoleConfig.Status.AccountRolesRef.InstallerRoleARN = role.RoleARN
+		}
+		if role.RoleName == fmt.Sprintf("%s-HCP-ROSA-Support-Role", config.Prefix) {
+			createRoles = false
+			scope.RosaRoleConfig.Status.AccountRolesRef.SupportRoleARN = role.RoleARN
+		}
+		if role.RoleName == fmt.Sprintf("%s-HCP-ROSA-Worker-Role", config.Prefix) {
+			createRoles = false
+			scope.RosaRoleConfig.Status.AccountRolesRef.WorkerRoleARN = role.RoleARN
+		}
+	}
+	if createRoles {
+		runtime.Reporter = (&rosa.Reporter{})
+		runtime.Logger = rosalogging.NewLogger()
+		runtime.AWSClient, err = aws.NewClient().Logger(runtime.Logger).Build()
+		if err != nil {
+			return fmt.Errorf("failed to create aws client: %w", err)
+		}
+
+		managedPolicies := true
+		isSharedVpc := config.SharedVPCConfig.VPCEndpointRoleARN != "" && config.SharedVPCConfig.RouteRoleARN != ""
+		err := accountroles.CreateHCPRoles(runtime, config.Prefix, managedPolicies, config.PermissionsBoundaryARN, ocm.Production, policies, config.Version, config.Path, isSharedVpc, config.SharedVPCConfig.RouteRoleARN, config.SharedVPCConfig.VPCEndpointRoleARN)
+		return err
+	}
+
+	return nil
+}
+
+func (r *ROSARoleConfigReconciler) createOIDCConfig(scope *scope.RosaRoleConfigScope, ocmClient *ocm.Client) error {
+	runtime := rosacli.NewRuntime()
+	var err error
+	runtime.Reporter = (&rosa.Reporter{})
+	runtime.OCMClient = ocmClient
+	runtime.Logger = rosalogging.NewLogger()
+	runtime.AWSClient, err = aws.NewClient().Logger(runtime.Logger).Build()
+	if err != nil {
+		return fmt.Errorf("failed to create aws client: %w", err)
+	}
+
+	runtime.Creator, err = runtime.AWSClient.GetCreator()
+	if err != nil {
+		return err
+	}
+
+	// userPrefix, region are used only for unmanaged OIDC config
+	oidcID, createErr := oidcconfig.CreateOIDCConfig(runtime, true, "", "")
+	if createErr != nil {
+		return fmt.Errorf("failed to Create OIDC config: %w", err)
+	}
+
+	scope.RosaRoleConfig.Status.OIDCID = oidcID
+	return createErr
+}
+
+func (r *ROSARoleConfigReconciler) deleteAccountRoles(ocmClient *ocm.Client, awsClient aws.Client, scope *scope.RosaRoleConfigScope) error {
+	roles := scope.RosaRoleConfig.Status.AccountRolesRef
+	config := scope.RosaRoleConfig.Spec.AccountRoleConfig
+	deleteHcpSharedVpcPolicies := config.SharedVPCConfig.VPCEndpointRoleARN != "" && config.SharedVPCConfig.RouteRoleARN != ""
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		return err
+	}
+
+	clusters, err := ocmClient.GetAllClusters(creator)
+	if err != nil {
+		return err
+	}
+
+	var err2, err3 error
```

Comment:
Please use descriptive names instead of numbers for the variable names.

Although it looks like we only ever return 1 error, so you could re-use `err`, rather than have separate variables and extra nil checks.